File: python/scrabble-score/scrabble_score.py
vals = [
    (["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"], 1),
    (["D", "G"], 2),
    (["B", "C", "M", "P"], 3),
    (["F", "H", "V", "W", "Y"], 4),
    (["K"], 5),
    (["J", "X"], 8),
    (["Q", "Z"], 10),
]
score_map = {}
for val in vals:
    for letter in val[0]:
        score_map[letter] = val[1]
def score(word):
    score = 0
    for letter in [*word]:
        score += score_map[letter.upper()]
    return score

File: ruby/scrabble-score/scrabble_score.rb
class Scrabble
  LETTERS = { D:2, G:2, B:3, C:3, M:3, P:3, F:4, H:4, V:4, W:4, Y:4, K:5, J:8, X:8, Q:10, Z:10 }.freeze
  def initialize(word)
    @word = word
  end

  def score
    @word.upcase.chars.map { |l| LETTERS[l.to_sym] || 1 }.sum
  end
end
